help/Readme.md

```diff
@@ -0,0 +1,114 @@
+- `PowerShell 5.0 or higher` or `PSCore` for linux distributes.
+- `Azure CLI ` - https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest
+
+#### Azure DevOps self-hosted pool requirements
+To connect to a temporary VM packer use WinRM or SSH connections on public IP interfaces.
+If you use a build agent located in an Azure subscription, please make sure that HTTPS/SSH ports are allowed for incoming/outgoing connections.
+In case of firewall restrictions, prohibiting connections from public addresses, private virtual network resources can be deployed and passed as arguments to the packer. This approach allows virtual machines to use private connections inside VLAN.
+
+#### Service principal
+Packer uses Service Principal to authorize in Azure infrastructure. To setup image-generation CI or use packer manually — SP with full read-write permissions for selected Azure subscription needed.
+Detailed instruction can be found [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+
+### Prepare environment and image deployment
+#### How to prepare Windows build agent
+You can use local machine as build agent, or deploy virtual machine in [Azure](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/quick-create-cli).
+
+Download `packer` from https://www.packer.io/downloads, or install it via Chocolately.
+```
+Choco-Install -PackageName packer
+```
+
+Install Azure CLI - https://docs.microsoft.com/ru-ru/cli/azure/install-azure-cli-windows?view=azure-cli-latest&tabs=azure-cli.
+```
+Invoke-WebRequest -Uri https://aka.ms/installazurecliwindows -OutFile .\AzureCLI.msi; Start-Process msiexec.exe -Wait -ArgumentList '/I AzureCLI.msi /quiet'; rm .\AzureCLI.msi
+```
+
+Download Virtual-Environments repository.
+```
+git clone https://github.com/actions/virtual-environments.git
+```
+
+Import [GenerateResourcesAndImage](https://github.com/actions/virtual-environments/blob/main/helpers/GenerateResourcesAndImage.ps1) script from `/helpers` folder, and run `GenerateResourcesAndImage` function in Powershell.
+
+```
+Import-Module C:\virtual-environments\helpers\GenerateResourcesAndImage.ps1
+
+GenerateResourcesAndImage -SubscriptionId {YourSubscriptionId} -ResourceGroupName "myTestResourceGroup" -ImageGenerationRepositoryRoot "C:\virtual-environments" -ImageType Ubuntu1604 -AzureLocation "East US"
+```
+Where:
+- `SubscriptionId` - The Azure subscription Id where resources will be created.
+- `ResourceGroupName` - The Azure resource group name where the Azure resources will be created.
+- `ImageGenerationRepositoryRoot` - The root path of the image generation repository source.
+- `ImageType` - The type of the image being generated. Valid options are: "Windows2016", "Windows2019", "Ubuntu1604", "Ubuntu1804", "Ubuntu2004".
+- `AzureLocation` - The location of the resources being created in Azure. For example "East US".
```

Comment:
- `AzureLocation` - Resources location in Azure. For example "East US".

Reply:
It's a copypaste from packer documentation, actually.